### PR TITLE
fix: jumpstart cache using sagemaker session s3 client

### PR DESCRIPTION
**Pull request by evakravi**
*Issue #, if available:*
https://github.com/aws/sagemaker-python-sdk/issues/4043

*Description of changes:*
Uses s3 client from SageMaker Session for accessing JumpStart cache. Also refactors JumpStart utility functions to include SageMaker Session argument

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/jumpstart/accessors.py [hunk 1]
```diff
@@ -90,11 +101,20 @@ def _get_manifest(region: str = JUMPSTART_DEFAULT_REGION_NAME) -> List[JumpStart
...
+            additional_kwargs.update({"s3_client": s3_client})
+
         cache_kwargs = JumpStartModelsAccessor._validate_and_mutate_region_cache_kwargs(
-            JumpStartModelsAccessor._cache_kwargs, region
+            {**JumpStartModelsAccessor._cache_kwargs, **additional_kwargs}, region
         )
         JumpStartModelsAccessor._set_cache_and_region(region, cache_kwargs)
-        return JumpStartModelsAccessor._cache.get_manifest()  # type: ignore
+        manifest = JumpStartModelsAccessor._cache.get_manifest()  # type: ignore
+        JumpStartModelsAccessor._cache_kwargs = old_cache_kwargs
```
Comment (chrstfu) [inline src/sagemaker/jumpstart/accessors.py:116]: Quick question - why are we setting the `cache_kwargs` value to the old value? Correct me if I'm wrong but would this make the `or new_cache_kwargs != JumpStartModelsAccessor._cache_kwargs` check above fail every time, resetting the cache each time?

Comment (evakravi) [reply src/sagemaker/jumpstart/accessors.py:116]: These methods are not intended to mutate the cache kwargs itself, there's another method for that `set_cache_kwargs`.

File: src/sagemaker/jumpstart/cache.py [hunk 2]
```diff
@@ -109,7 +111,7 @@ def __init__(
             if s3_bucket_name is None
             else s3_bucket_name
         )
-        self._s3_client = (
+        self._s3_client = s3_client if s3_client else (
```
Comment (JGuinegagne) [inline src/sagemaker/jumpstart/cache.py:114]: non-blocking: `s3_client or boto3.client()`

File: src/sagemaker/jumpstart/factory/estimator.py [hunk 3]
```diff
@@ -381,7 +385,7 @@ def _add_region_to_kwargs(kwargs: JumpStartKwargs) -> JumpStartKwargs:
 
 def _add_sagemaker_session_to_kwargs(kwargs: JumpStartKwargs) -> JumpStartKwargs:
     """Sets session in kwargs based on default or override, returns full kwargs."""
-    kwargs.sagemaker_session = kwargs.sagemaker_session or Session()
+    kwargs.sagemaker_session = kwargs.sagemaker_session or DEFAULT_JUMPSTART_SAGEMAKER_SESSION
```
Comment (JGuinegagne) [inline src/sagemaker/jumpstart/factory/estimator.py:388]: sanity-check: wouldn't this default all the sessions to `us-west-2` ?

File: src/sagemaker/jumpstart/utils.py [hunk 4]
```diff
@@ -397,6 +397,7 @@ def verify_model_region_and_return_specs(
     region: str,
     tolerate_vulnerable_model: bool = False,
     tolerate_deprecated_model: bool = False,
+    sagemaker_session: Session = constants.DEFAULT_JUMPSTART_SAGEMAKER_SESSION,
```
Comment (JGuinegagne) [inline src/sagemaker/jumpstart/utils.py:400]: same question here

**Review by JGuinegagne — COMMENTED** [review]
I haven't looked at the unit tests yet, but couple high level questions

File: src/sagemaker/jumpstart/factory/estimator.py (hunk 3 above)
Comment (evakravi) [reply src/sagemaker/jumpstart/factory/estimator.py:388]: Only if they don't have any region set in the AWS config, which is usually not the case or only done in error.

**Review by knikure — APPROVED** [review]

**Pull request merged**